src/ast_fwd_decl.hpp

```diff
@@ -385,6 +393,21 @@ namespace Sass {
     }
   };
 
+  template <class T>
+  bool SetsAreEqual(const T& lhs, const T& rhs) {
+    if (lhs.size() != rhs.size()) return false;
+    auto lit = lhs.begin(), lend = lhs.end();
+    auto rit = rhs.begin() /* , rend = rhs.end() */;
+    return std::equal(lit, lend, rit, CompareNodes());
+  }
+
+  template <class T>
+  bool SetContains(const T& lhs, const T& rhs) {
```

Comment:
How about using hash maps/sets throughout (`std::unordered_map`)? They faster and easier to get right